pxr/imaging/hd/instanceRegistry.h

```diff
@@ -59,46 +64,44 @@ class HdInstance {
         ValueType value;
         int recycleCounter;
     };
-    typedef tbb::concurrent_unordered_map<KeyType, ValueHolder> Dictionary;
-
-    typedef std::mutex RegistryMutex;
-    typedef std::unique_lock<RegistryMutex> RegistryLock;
 
     HdInstance() = delete;
 
     /// Construct an instance holding a registry lock, representing a value
     /// held in a registry container.
-    explicit HdInstance(KeyType const &key,
-                        ValueType const &value,
-                        RegistryLock &&registryLock,
-                        Dictionary *container)
+    explicit HdInstance(HdInstanceKey           key,
+                        ValueType const         &value,
+                        HdInstanceRegistryLock  &&registryLock,
+                        REGISTRY                *registry)
         : _key(key)
         , _value(value)
         , _registryLock(std::move(registryLock))
-        , _container(container)
+        , _registry(registry)
         , _isFirstInstance(!bool(_value))
     { }
 
     /// Construct an instance with no lock or registry container. This
     /// is used to present a consistent interface to clients in cases
     /// where shared resource registration is disabled.
-    explicit HdInstance(KeyType const &key)
+    explicit HdInstance(HdInstanceKey key)
```

Comment:
We know that the key is always just an integer, so it's better to pass it around by value.